File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int,int>mp;
        int ans = 0;
        int length = 0;
        for(int i=0; i<nums.size();i++)
        {
            mp[nums[i]]++;
        }
        for(int i=0; i<nums.size();i++)
        {
            if(mp.find(nums[i]-1)==mp.end())
            {
                length = 1;
                int num = nums[i]+1;
                while(mp.find(num)!=mp.end())
                {
                    length++;
                    num++;

                }
                 ans = max(ans, length);
            }
        }

       
        return ans;
        
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        count = 0
      
       
        for n in num_set:
            if n - 1 not in num_set:
                length = 1
                while length + n in num_set:
                    length +=1
                count = max(length, count)
        return count